components/integrations/view/coordinape/IntegrationForm.js

```diff
@@ -0,0 +1,93 @@
+import { useState } from 'react'
+
+import useWeb3 from '../../../hooks/useWeb3'
+import useCommunity from '../../../hooks/useCommunity'
+import { parseCoordinapeCSV } from '../../../../util/csv'
+import Button from '../../../ui/Button'
+
+const CoordinapeIntegrationForm = () => {
+  const [files, setFiles] = useState([])
+  const [isProcessing, setIsProcessing] = useState(false)
+  const { bearerToken } = useWeb3()
+  const { community } = useCommunity()
+
+  if (!community) return <h2>Loading community...</h2>
+
+  const handleFileChange = ({ currentTarget: { files: inputFiles } }) => {
+    const csvFiles = []
+    Array.from(inputFiles).forEach(file => {
+      const { type } = file
+      if (type === 'text/csv') {
+        csvFiles.push(file)
+      }
+    })
+    setFiles(csvFiles)
+  }
+
+  const handleProcessFiles = async () => {
+    setIsProcessing(true)
+    Array.from(files).map(async file => {
+      try {
+        const text = await file.text()
+        const coordinapeData = parseCoordinapeCSV(text)
+        const response = await fetch('/api/triggers/coordinape', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Token ${bearerToken}`,
+          },
+          body: JSON.stringify({
+            name: file.name,
+            contributions: coordinapeData,
+          }),
```

Comment:
overkill to move into hook?

Reply:
Probably not in this case, as we shouldn’t have a lot of these where we are uploading CSVs (I hope).